components/clarity-repl/src/repl/datastore.rs

```diff
@@ -319,193 +325,316 @@ impl ClarityBackingStore for Datastore {
-        height_at_chain_tip.insert(id, 0);
-
-        let mut sortition_lookup = HashMap::new();
-        sortition_lookup.insert(sortition_id, id);
-
-        let mut consensus_hash_lookup = HashMap::new();
-        consensus_hash_lookup.insert(genesis_block.consensus_hash, sortition_id);
-
-        let mut store = HashMap::new();
-        store.insert(id, genesis_block);
-
-        let mut block_id_lookup = HashMap::new();
-        block_id_lookup.insert(id, id);
-
-        let mut id_height_map = HashMap::new();
-        id_height_map.insert(id, 0);
+        let genesis_block = StacksBlockInfo {
+            block_header_hash: BlockHeaderHash([0x00; 32]),
+            burn_block_header_hash: first_burn_block_header_hash,
+            consensus_hash: ConsensusHash([0x00; 20]),
+            vrf_seed: VRFSeed([0x00; 32]),
+            stacks_block_time: genesis_time + 10,
+        };
 
-        BurnDatastore {
-            store,
+        let sortition_lookup = HashMap::from([(sortition_id, id)]);
+        let consensus_hash_lookup = HashMap::from([(genesis_block.consensus_hash, sortition_id)]);
+        let burn_blocks = HashMap::from([(first_burn_block_header_hash, genesis_burn_block)]);
+        let stacks_blocks = HashMap::from([(id, genesis_block)]);
+
+        Datastore {
+            genesis_id: id,
+            burn_chain_height: 0,
+            burn_blocks,
+            stacks_chain_height: 0,
+            stacks_blocks,
             sortition_lookup,
             consensus_hash_lookup,
-            block_id_lookup,
-            open_chain_tip: id,
-            current_chain_tip: id,
-            chain_height: 0,
-            height_at_chain_tip,
             current_epoch: StacksEpochId::Epoch2_05,
             current_epoch_start_height: 0,
             constants,
-            genesis_time,
         }
     }
 
     pub fn get_current_epoch(&self) -> StacksEpochId {
         self.current_epoch
     }
 
-    pub fn get_current_block_height(&self) -> u32 {
-        self.chain_height
+    pub fn get_current_stacks_block_height(&self) -> u32 {
+        self.stacks_chain_height
     }
-    pub fn advance_chain_tip(&mut self, count: u32) -> u32 {
-        let cur_height = self.chain_height;
-        let current_lookup_id = *self
+
+    pub fn get_current_burn_block_height(&self) -> u32 {
+        self.burn_chain_height
+    }
+
+    fn build_next_stacks_block(&self, clarity_datastore: &ClarityDatastore) -> StacksBlockInfo {
+        let burn_chain_height = self.burn_chain_height;
+        let stacks_block_height = self.stacks_chain_height;
+
+        let last_stacks_block = self
+            .stacks_blocks
+            .get(&clarity_datastore.current_chain_tip)
+            .expect("current chain tip missing in stacks block table");
+        let last_burn_block = self
+            .burn_blocks
+            .get(&height_to_burn_block_header_hash(burn_chain_height))
+            .expect("burn block missing in burn block table");
```

Comment:
An area to consider some improvement, if is the case:

. using `expect()` after these lookups could panic the program if the keys are not found.